lib/elixir/lib/code/fragment.ex

```diff
@@ -656,6 +665,9 @@ defmodule Code.Fragment do
           {{:dot, _, [_ | _]} = dot, offset} ->
             build_surround(dot, reversed, line, offset)
 
+          {{:local_or_var, acc}, offset} when keyword? ->
+            build_surround({:keyword, acc}, reversed, line, offset)
```

Comment:
I don't think `keyword` should be returned here. It's already used for reserved words `do end after else catch rescue fn true false nil`

Reply:
Got it, I misunderstood what `keyword` meant in this context.
`keyword_atom` / `keyword_key`?

Reply:
`Macro.inspect_atom` uses `key`. Maybe that's fine? Or we use `atom_key` or `keyword_key`.

Reply:
Either sounds good, I went with `:key` for the consistency.

Last question: in the case of `do:`, should it be a `:key` or a `:keyword`?
It would be `:key` with the current PR implementation, but I'm not sure which would make sense from a LSP perspective @lukaszsamson

Reply:
Cf [7da1991](https://github.com/elixir-lang/elixir/pull/13843/commits/7da1991516493eded28a85411d6107722124b4c3)

Reply:
`do:` should be `:key` IMO.

Reply:
It's OK, let's have `do:` as `key` and `do` as `keyword`.

Reply:
Great, thanks gents will merge as is then 🚀

Reply:
I just realized: should we call out this "edge case" in the docs?

Reply:
I don't think it is an edge case, honestly. It is how the grammar behaves everywhere. :) So there is no need imo!